src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -775,24 +774,27 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
 
     if (hasCredit) {
       // Settlement is via the Customer's cash balance, with no client-side provision of a Payment Method.
-      this.donationService.finaliseCashBalancePurchase(this.donation).subscribe(
-        (donation) => {
+      this.donationService.finaliseCashBalancePurchase(this.donation).subscribe({
+        next: (donation) => {
           this.matomoTracker.trackEvent(
             'donate',
             'stripe_customer_balance_payment_success',
             `Stripe Intent expected to auto-confirm for donation ${donation.donationId} to campaign ${donation.projectId}`,
           );
           this.exitPostDonationSuccess(donation, 'donation-funds');
         },
-        (error: HttpErrorResponse) => {
+        error: (response: HttpErrorResponse) => {
+          // I think this is the path to a detailed message in MatchBot `ActionError`s.
+          const errorMsg = response.error?.error?.description;
+
           this.matomoTracker.trackEvent(
             'donate_error',
             'stripe_customer_balance_payment_error',
-            error.message ?? '[No message]',
+            errorMsg ?? '[No message]',
           );
-          this.stripeError = `Cash balance application failed: ${error.message}`;
+          this.stripeError = 'Your donation has not been processed as it seems you have insufficient funds. Please refresh the page to see your remaining balance.';
```

Comment:
We're not 100% sure at this point that the two things are related, are we? So maybe better in separate sentences?

```suggestion
          this.stripeError = 'Your donation has not been processed. It seems you have insufficient funds. Please refresh the page to see your remaining balance.';
```

Reply:
I think that we're fairly sure – I don't understand how the new syntax would be more accurate in any scenario we know about, so I'd favour keeping the requested copy from Jira for now.